### PR TITLE
Actions workflow fixes

### PR DESCRIPTION
**Pull request by KingArrkinian**
Changes:
- Datasets location from branch:develop to any branch (defaults to master)
- Upload_datasets_on_schedule renamed to upload_datasets
- Modified the Slack bot to only be called on pull requests
- Many enhancements on https://github.com/KingArrkinian/slack-pr-open-notification-action@master

Help needed:
- The HAL Plugins error has returned despite the code not changing (3 ls were removed, re-added, and it didn't seem to change a thing)
- This morning: https://github.com/KingArrkinian/openeb/runs/3556417182?check_suite_focus=true
- This afternoon: https://github.com/KingArrkinian/openeb/runs/3560534840?check_suite_focus=true

### CODE REVIEW

File: .github/workflows/all_jobs.yaml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-name: All Build on PR
+name: All Build
```
Comment (cbernard-psee) [inline .github/workflows/all_jobs.yaml:1]: What is the rationnal to remove "on PR"
It seems PR criteria is still in place below using "github.event_name == 'pull_request' "

File: .github/workflows/all_jobs.yaml [hunk 2]
```diff
@@ -188,34 +182,12 @@ jobs:
...
-      #     && "${{ needs.job3.outputs.job.status }}" == "Success" ]]; then
-      #     $results="Success"
-      #     fi
-
       - name: Slack PR hook test
-        uses: kingarrkinian/slack-pr-open-notification-action@no_fork_option
+        uses: kingarrkinian/slack-pr-open-notification-action@master
         env:
           SLACK_WEBHOOK_URL: ${{ secrets.OPENEB_GITHUB_PROPHESEE_SLACK_WEBHOOK }}
-          PULL_REQUEST_NUMBER : ${{ github.event.pull_request.number }}
```
Comment (cbernard-psee) [inline .github/workflows/all_jobs.yaml:208]: Slack msg capability will be enhanced in another PR, correct?

File: .github/staging/windows_push.yaml [hunk 3]
```diff
@@ -50,11 +50,8 @@ jobs:
     - name: Download HAL Plugins and move DLLs into /build
```
Comment (cbernard-psee) [inline .github/staging/windows_push.yaml:50]: skipping the review of this file since there is a restriction on windows stage for now . WIP :-)

File: .github/workflows/all_jobs.yaml (hunk 1 above)
Comment (cbernard-psee) [reply .github/workflows/all_jobs.yaml:1]: clarified by IM

**Review by cbernard-psee — APPROVED** [review]

File: .github/workflows/all_jobs.yaml (hunk 2 above)
Comment (KingArrkinian) [reply .github/workflows/all_jobs.yaml:208]: Code now automatically finds these github variables. These statements are now not necessary.

File: .github/staging/windows_push.yaml (hunk 3 above)
Comment (KingArrkinian) [reply .github/staging/windows_push.yaml:50]: The windows stage is being tested on my github_actions-test branch. Once results are secured, another PR will be sent along.

**Review by nbourdis-psee — APPROVED** [review]
